runtime/src/common/sgx/pcs/quote.rs

```diff
@@ -0,0 +1,914 @@
+            }
+        }
+
+        // Parse the report body based on the TEE.
+        match quote.header.tee_type() {
+            TeeType::SGX => {
+                // Parse SGX report body.
+                let mut report_body = Vec::with_capacity(SgxReport::UNPADDED_SIZE);
+                report_body.extend(quote.report_body());
+                report_body.resize_with(SgxReport::UNPADDED_SIZE, Default::default);
+                let report_body =
+                    SgxReport::try_copy_from(&report_body).ok_or(Error::MalformedReport)?;
+
+                // Disallow debug enclaves, if we are in production environment and disallow production
+                // enclaves, if we are in debug environment.
+                let is_debug = report_body
+                    .attributes
+                    .flags
+                    .contains(AttributesFlags::DEBUG);
+                let allow_debug = option_env!("OASIS_UNSAFE_ALLOW_DEBUG_ENCLAVES").is_some();
+                if is_debug && !allow_debug {
+                    return Err(Error::DebugEnclave);
+                } else if !is_debug && allow_debug {
+                    return Err(Error::ProductionEnclave);
+                }
+
+                Ok(VerifiedQuote {
+                    report_data: report_body.reportdata.to_vec(),
+                    identity: EnclaveIdentity {
+                        mr_enclave: MrEnclave::from(report_body.mrenclave.to_vec()),
+                        mr_signer: MrSigner::from(report_body.mrsigner.to_vec()),
+                    },
+                    timestamp,
+                })
+            }
+            TeeType::TDX => {
+                // Parse TDX report body.
+                let report_body = TdxReport::parse(quote.report_body())?;
+
+                // Disallow debug TDs, if we are in production environment and disallow production
+                // TDs, if we are in debug environment.
+                let is_debug = report_body.td_attributes.contains(TdxAttributes::DEBUG);
+                let allow_debug = option_env!("OASIS_UNSAFE_ALLOW_DEBUG_ENCLAVES").is_some();
+                if is_debug && !allow_debug {
+                    return Err(Error::DebugEnclave);
+                } else if !is_debug && allow_debug {
+                    return Err(Error::ProductionEnclave);
+                }
+
+                // Verify report against TDX policy.
+                let tdx_policy = policy.tdx.as_ref().ok_or(Error::TeeTypeNotAllowed)?;
+                tdx_policy.verify(&report_body)?;
+
+                Ok(VerifiedQuote {
+                    report_data: report_body.report_data.to_vec(),
+                    identity: report_body.as_enclave_identity(),
+                    timestamp,
+                })
+            }
+        }
+    }
+}
+
+/// An enclave quote.
+#[derive(Debug)]
+pub struct Quote<'a> {
+    raw: Cow<'a, [u8]>,
+    header: QuoteHeader<'a>,
+    report_body: Cow<'a, [u8]>,
+    signature: Cow<'a, [u8]>,
```

Comment:
Well, in this case `enclave_report` (used previously) would be incorrect as for TDX the reports are TD reports not enclave reports. So `report_body` seems more general.

Reply:
You are right, `report_body` is fine for `enclave_report` and  `td_report`. Since this was inspired by Fortanix, ignore this comment (and some other comments as well).